lib/db/schemaupdater.go

```diff
@@ -563,6 +563,12 @@ func (db *schemaUpdater) updateSchemato10(_ int) error {
 	for _, folderStr := range db.ListFolders() {
 		folder := []byte(folderStr)
 
+		// Update the existing metadata if it can be read, otherwise
+		// it will get recreated later in normal operations.
+		meta := newMetadataTracker()
+		metaErr := meta.fromDB(db.Lowlevel, []byte(folder))
+		meta.resetAll(protocol.GlobalDeviceID) // Is rewritten in updateGlobal
```

Comment:
I believe this is wrong: Updating global will do nothing except updating the `Deleted` field of in global list entries, as the versions won't change (thus no change in need or global counts). I probably need to not touch global  and recalc needs only. And I definitely want to write a test for the transition :)

Reply:
It is good (wrote the test) - a few lines above I empty the global list (because it wouldn't do anything otherwise, not even update the deleted field as the version doesn't change).